Bottom: 590d78abca6568895476bf489103bd40362432d6
Top:    24f89871fe8fd09e2aabddde3f385388d49b442c
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 13:54:07 +0800

autofs-5.0.7 - fix potential null dereference in lookup_mount()

Setting rv to CHE_OK by default could allow a null map entry dereference in
lookup_mount() in several modules.


---

diff --git a/modules/lookup_file.c b/modules/lookup_file.c
index 2836996..58663c6 100644
--- a/modules/lookup_file.c
+++ b/modules/lookup_file.c
@@ -1131,7 +1131,7 @@ do_cache_lookup:
 				       mapent, ctxt->parse->context);
 	if (ret) {
 		time_t now = time(NULL);
-		int rv = CHE_OK;
+		int rv = CHE_FAIL;
 
 		cache_writelock(mc);
 		me = cache_lookup_distinct(mc, key);
diff --git a/modules/lookup_ldap.c b/modules/lookup_ldap.c
index 6e06e02..3371b3f 100644
--- a/modules/lookup_ldap.c
+++ b/modules/lookup_ldap.c
@@ -3012,7 +3012,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 				       mapent, ctxt->parse->context);
 	if (ret) {
 		time_t now = time(NULL);
-		int rv = CHE_OK;
+		int rv = CHE_FAIL;
 
 		/* Record the the mount fail in the cache */
 		cache_writelock(mc);
diff --git a/modules/lookup_sss.c b/modules/lookup_sss.c
index 5c2ed0a..013d3b4 100644
--- a/modules/lookup_sss.c
+++ b/modules/lookup_sss.c
@@ -673,7 +673,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 				       mapent, ctxt->parse->context);
 	if (ret) {
 		time_t now = time(NULL);
-		int rv = CHE_OK;
+		int rv = CHE_FAIL;
 
 		/* Record the the mount fail in the cache */
 		cache_writelock(mc);
diff --git a/modules/lookup_yp.c b/modules/lookup_yp.c
index a716e1f..72a8baa 100644
--- a/modules/lookup_yp.c
+++ b/modules/lookup_yp.c
@@ -699,7 +699,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 					       mapent, ctxt->parse->context);
 		if (ret) {
 			time_t now = time(NULL);
-			int rv = CHE_OK;
+			int rv = CHE_FAIL;
 
 			cache_writelock(mc);
 			me = cache_lookup_distinct(mc, key);
